distributed/scheduler.py

```diff
@@ -2675,7 +2684,8 @@ def _transition_processing_erred(
 
         if not ts.erred_on:
             ts.erred_on = set()
-        ts.erred_on.add(worker)
+        if worker:
+            ts.erred_on.add(worker)
```

Comment:
I think the type in the function declaration should change to `worker: str | None`?

Reply:
My bad, I didn't clean this up properly.